snuba/query/__init__.py

```diff
@@ -256,7 +256,11 @@ def _transform_expressions_impl(
         """
         raise NotImplementedError
 
-    def transform_expressions(self, func: Callable[[Expression], Expression]) -> None:
+    def transform_expressions(
+        self,
+        func: Callable[[Expression], Expression],
+        skip_transform_condition: bool = False,
```

Comment:
I am not convinced by this change. It seems ad-hoc instead of being a reasonable generic API.
What about providing a list of expressions to change instead of a single flag to ignore?
```
transform_expression(self, func): # transforms everything
     transform_expressions_clauses(self, func, [])

transform_expressions_clauses(self, func, clauses_to_skip: Set[ClauseEnum]
```

Reply:
I'm not really against providing the generic solution at all, but it occurred to me after working on these 2 processors that skipping just the condition is actually the main one that is going to be useful for us. I'm not sure we need the general solution yet, and I'm a little hesitant to just build it until we do.

Reply:
ok, let's do this way, the next time we need to filter a clause we replace this with a generic method and adapt all the cases where this is used. Are you ok with that ?

Reply:
👍